doc/topics/the-event-loop-timers-and-nexttick

```diff
@@ -0,0 +1,174 @@
+# Overview of the Event Loop, Timers, and `process.nextTick()`
+
+The Following diagram shows a simplified overview of the event loop's order of operations.
+
+       ┌───────────────────────┐
+    ┌─>│        timers         │
+    │  └──────────┬────────────┘
+    │  ┌──────────┴────────────┐
+    │  │   pending callbacks   │
+    │  └──────────┬────────────┘      ┌───────────────┐
+    │  ┌──────────┴────────────┐      │   incoming:   │
+    │  │         poll          │<─────│  connections, │
+    │  └──────────┬────────────┘      │   data, etc.  │
+    │  ┌──────────┴────────────┐      └───────────────┘
+    └──│     setImmediate      │
+       └───────────────────────┘
+
+note: each box will be referred to as a "phase" of the event loop.
+
+*There is a slight discrepancy between the Windows and the Unix/Linux
+implementation, but that's not important for this demonstration.  The most
+important parts are here.  There are actually seven or eight steps, but the
+ones we care about--ones that Node actually uses are these four.*
```

Comment:
> I don't think so. There are other minor steps inbetween these four provided, but as I understand, they are not particularly notable to an end user. (i.e. how this interacts with the microtask queue.)

I'd lean towards documenting microtask queue behavior if it's feasible to do so here — I've definitely found that information useful in the past when dealing with builtin promises.